File: Week 1/id_295/LeetCode_1_295.java
import java.util.HashMap;

public class TwoSum {
    class Solution {
        public int[] twoSum(int[] nums, int target) {
            HashMap<Integer, Integer> map = new HashMap<>();
            for (int index = 0; index < nums.length; index++) {
                int complement = target - nums[index];
                if (map.containsKey(complement)) {
                    return new int[]{map.get(complement), index};
                }
                map.put(nums[index], index);
            }
            throw new IllegalArgumentException("No answer");
        }
    }
}

File: Week 1/id_295/Python/1_3/LeetCode_1_295.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for index, num in enumerate(nums):
            if num in d:
                return [d[num], index]
            d[target - num] = index


def test_answer():
    assert Solution().twoSum([2, 7, 11, 15], 9) == [0, 1]
